### PR TITLE
Fix a couple tests not running on CI

### PR DESCRIPTION
**Pull request by blampe**
I noticed as part of https://github.com/pulumi/pulumi-kubernetes/pull/2953 that the `tests/convert` and `tests/provider` directories aren't currently getting exercised in CI, and as a result one of them is failing on master:

```
❯ go test -short -v -cover -timeout 2h ./...
=== RUN   TestTerraformConvert
    convert_test.go:37: output: Converting from terraform...
        Converting to yaml...
        warning: using pulumi-language-yaml from $PATH at /opt/homebrew/bin/pulumi-language-yaml
    convert_test.go:45:
        	Error Trace:	/Users/bryce/src/pulumi-kubernetes/tests/convert/convert_test.go:45
        	Error:      	Not equal:
        	            	expected: "resources:\n  nginx:\n    type: kubernetes:apps/v1:Deployment\n    properties:\n      metadata:\n        name: scalable-nginx-example\n        labels:\n          app: ScalableNginxExample\n      spec:\n        replicas: 2\n        selector:\n          matchLabels:\n            app: ScalableNginxExample\n        template:\n          metadata:\n            labels:\n              app: ScalableNginxExample\n          spec:\n            containers:\n              - image: nginx:1.7.8\n                name: example\n                ports:\n                  - containerPort: 80\n                resources:\n                  limits:\n                    cpu: '0.5'\n                    memory: 512Mi\n                  requests:\n                    cpu: 250m\n                    memory: 50Mi\n  nginxResource:\n    type: kubernetes:core/v1:Service\n    properties:\n      metadata:\n        name: nginx-example\n      spec:\n        selector:\n          app: ${nginx.spec.template.metadata.labels.app}\n        ports:\n          - nodePort: 30201\n            port: 80\n            targetPort: 80\n        type: NodePort\n"
        	            	actual  : "resources:\n  nginx:\n    type: kubernetes:apps/v1:Deployment\n    properties:\n      metadata:\n        name: scalable-nginx-example\n        labels:\n          app: ScalableNginxExample\n      spec:\n        replicas: 2\n        selector:\n          matchLabels:\n            app: ScalableNginxExample\n        template:\n          metadata:\n            labels:\n              app: ScalableNginxExample\n          spec:\n            containers:\n              - image: nginx:1.7.8\n                name: example\n                ports:\n                  - containerPort: 80\n                resources:\n                  limits:\n                    cpu: '0.5'\n                    memory: 512Mi\n                  requests:\n                    cpu: 250m\n                    memory: 50Mi\n  nginxService:\n    type: kubernetes:core/v1:Service\n    name: nginx\n    properties:\n      metadata:\n        name: nginx-example\n      spec:\n        selector:\n          app: ${nginx.spec.template.metadata.labels.app}\n        ports:\n          - nodePort: 30201\n            port: 80\n            targetPort: 80\n        type: NodePort\n"

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -30,4 +30,5 @@
        	            	                     memory: 50Mi
        	            	-  nginxResource:
        	            	+  nginxService:
        	            	     type: kubernetes:core/v1:Service
        	            	+    name: nginx
        	            	     properties:
        	Test:       	TestTerraformConvert
--- FAIL: TestTerraformConvert (1.04s)
FAIL
coverage: [no statements]
FAIL	github.com/pulumi/pulumi-kubernetes/tests/v4/convert	1.179s
FAIL
```

This PR moves `tests/provider` and `tests/convert` under `sdk/java`, which is currently executed in CI but doesn't contain any tests. We also fix the convert test along the way.

"sdk/java" is admittedly an odd namespace to put these under, but our CI targets don't give us much flexibility at the moment, and at the end of the day it's just an arbitrary way of sharding things.

### CODE REVIEW

File: tests/sdk/java/convert_test.go [hunk 1]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package convert
+package test
```
Comment (rquitales) [inline tests/sdk/java/convert_test.go:15]: Could we add a code comment somewhere to describe why we are using this namespace for the convert tests, and perhaps a todo to restructure this in the future.

**Review by rquitales — APPROVED** [review]
LGTM. Might be worthwhile to create a tracker issue to restructure our tests in the future,

File: tests/sdk/java/convert_test.go (hunk 1 above)
Comment (blampe) [reply tests/sdk/java/convert_test.go:15]: Added a `doc.go` with a link to https://github.com/pulumi/ci-mgmt/issues/676 which I think best captures the spirit of what we need to do here.

**Comment by blampe** [comment]
Going to merge this - I've got a bunch of YAML tests starting to pile up and I definitely want to run them on CI!

As mentioned in chat, I'd prefer we live with the slightly awkward naming instead of baking an exception for Kubernetes into ci-mgmt. We can and should instead re-think how we do this sharding in the first place.

**Pull request merged**